File: web/src/app/models/request/registrar-venda-request.ts
export interface RegistrarVendaRequest {
  consumidor: string;
  formaPagamento: string;
  acrescimo: number;
  itens: RegistrarVendaItemRequest[];
}

export interface RegistrarVendaItemRequest {
  produtoId: string;
  codigoProduto: string;
  descricaoProduto: string;
  quantidade: number;
  precoUnitario: number;
  descontoValor: number;
  descontoPercentual: number;
}

File: api/Feak/Feak.SistemaFinanceiro.DomainService/DTOs/Request/RegistrarVendaRequest.cs
using System.ComponentModel.DataAnnotations;

namespace DomainService.DTOs.Request;

public class RegistrarVendaRequest
{
    [Required(ErrorMessage = "Consumidor obrigatório")]
    public string Consumidor { get; set; } = string.Empty;

    [Required(ErrorMessage = "Forma de pagamento obrigatória")]
    public string FormaPagamento { get; set; } = string.Empty;

    public decimal Acrescimo { get; set; }

    [MinLength(1, ErrorMessage = "Informe pelo menos um item")]
    public List<RegistrarVendaItemRequest> Itens { get; set; } = new();
}